### PR TITLE
Fix an issue with simulators being re-created when noReset is specified

### PR DESCRIPTION
**Pull request by fuzzybinary**
If a platformVersion is not specified, but noReset is specified, the driver creates a simulator for the specific session but does not clean it up. This adds a check for noReset and gets a simulator based on a the detected platform version in that case.

This fixes https://github.com/appium/appium/issues/9134

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
Thanks for the PR. Can you please sign the CLA?

File: lib/driver.js [hunk 1]
```diff
@@ -642,6 +642,15 @@ class XCUITestDriver extends BaseDriver {
                `This may cause problems if a simulator does not exist for this platform version.`);
       this.opts.platformVersion = this.iosSdkVersion;
     }
+
+    if (this.opts.noReset) {
```
Comment (jlipps) [inline lib/driver.js]: i'm curious why this change is restricted to the case where `noReset` is true. why wouldn't we want to use an existing sim with the default platformVersion in _any_ case we can?

Comment (fuzzybinary) [reply lib/driver.js]: This is because I'm assuming that in cases where noReset is false you do not want to just use the default simulators matchinging the capabilities, and would rather have an isolated simulator.

If this isn't the case, we can actually I think make this even simpler by moving lines 649-644 where it gets the capabilities up to above the first "getExistingSim."  I did that as well and it works, but I was afraid it might break existing functionality.

Comment (jlipps) [reply lib/driver.js]: Yes, that's a good thought to avoid changing the existing behavior. Hmm, what do you think @imurchie?

Comment (fuzzybinary) [reply lib/driver.js]: Any further thoughts on this? I can modify if needed.

Comment (imurchie) [reply lib/driver.js]: That sounds fine to me.

**Comment by imurchie** [comment]
Published in `appium-xcuitest-driver@2.47.1`
